include/crocoddyl/core/utils/exception.hpp

```diff
@@ -0,0 +1,32 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_CORE_UTILS_EXCEPTION_HPP_
+#define CROCODDYL_CORE_UTILS_EXCEPTION_HPP_
+
+#include <exception>
+#include <iostream>
+#include <sstream>
+
+namespace crocoddyl {
+
+class CrocoddylException : public std::exception {
```

Comment:
Why are we introducing a new exception type `CrocoddylException` rather than using `invalid_argument` for instance?

Reply:
Because we need to register the exceptions in the Python world. Otherwise, when the exception is triggered in Python, you will get message saying unidentified c++ exception.

This is the recommended approach in boost python

Reply:
I have been throwing std::invalid_argument and Boost Python has correctly translated them without this message. This is how it's done in Pinocchio for instance

Reply:
Done, thanks for the tip